File: Introductions/The Rust Programming Language/embed/bindings/embed.c
// Clang or GCC
//   $ clang -Wall embed.c -L../target/release/ -lembed -o embed
//   $ gcc -Wall embed.c -L../target/release/ -lembed -o embed

extern void process();

int main(int argc, char **argv) {
    process(); // call Rust function
    return 0;
}

File: Introductions/The Rust Programming Language/embed/bindings/embed.rb
# $ gem install ffi
# $ ruby embed.rb

require 'ffi'

module Hello
  extend FFI::Library
  ffi_lib '../target/release/libembed.dylib' #=> for Mac
  #ffi_lib '../target/release/libembed.so' #=> for Linux
  attach_function :process, [], :void
end

Hello.process

puts 'done!'
